config/aaa.py

```
@@ -243,16 +249,66 @@ def authtype(db, ctx, type):
+                secure_cipher.register("TACPLUS", "TACPLUS|global")
+                passwd = getpass.getpass()
+                # Set new password for encryption
+                secure_cipher.set_feature_password("TACPLUS", passwd)
+            else:
+                # Check if password rotation is enabled
+                if rotate:
+                    passwd = getpass.getpass()
+                    # Rotate password for TACPLUS feature and re-encrypt the secret
+                    secure_cipher.rotate_feature_passwd("TACPLUS", passwd)
+                    return
+            b64_encoded = secure_cipher.encrypt_passkey("TACPLUS", secret)
+            if b64_encoded is not None:
+                # Update key_encrypt flag
+                add_table_kv('TACPLUS', 'global', 'key_encrypt', True)
+                add_table_kv('TACPLUS', 'global', 'passkey', b64_encoded)
+            else:
+                # Deregister feature with Security Cipher module
+                secure_cipher.deregister("TACPLUS", rotate_tacplus_key)
+                click.echo('Passkey encryption failed')
+                return
+        except (EOFError, KeyboardInterrupt):
+            # Deregister feature with Security Cipher module
+            secure_cipher.deregister("TACPLUS", "TACPLUS|global")
+            add_table_kv('TACPLUS', 'global', 'key_encrypt', False)
+            click.echo('Input cancelled')
+            return
+        except Exception as e:
+            # Deregister feature with Security Cipher module
+            secure_cipher.deregister("TACPLUS", "TACPLUS|global")
+            add_table_kv('TACPLUS', 'global', 'key_encrypt', False)
+            click.echo('Unexpected error: %s' % e)
+            return
     else:
-        click.echo('Argument "secret" is required')
+        if secure_cipher.is_key_encrypt_enabled("TACPLUS", "global"):
+            secure_cipher.deregister("TACPLUS", "TACPLUS|global")
```

so if we run without --encrypt, we will remove the associated key encryption password and set `key_encrypt` to false, since the user is no longer wanting encryption. this seems correct.

What about the --rotate option though? if a user passes --rotate without --encrypt, what then? we should at least throw an error in that case, rather than removing the encryption.